### ISSUE TITLE
Topcat Caveat not explicit enough

### ISSUE THREAD

**Issue by odemangeon** [Homebrew/homebrew-science#3325]
I've installed Topcat with homebrew and the installation goes well except that it produces me a Caveats has shown here:

```
$brew info topcat
homebrew/science/topcat: stable 4.0-1
http://www.star.bris.ac.uk/~mbt/topcat/
/usr/local/Cellar/topcat/4.0-1 (3 files, 25.2M) *
  Built from source
From: https://github.com/Homebrew/homebrew-science/blob/master/topcat.rb
==> Caveats
The Java JAR files are installed to
  /usr/local/Cellar/topcat/4.0-1/share/java
```

Usually caveats are a bit more explicit and tells what you have to do. I am not very experienced with homebrew and here I have not idea what to do. So I did nothing but when I launch topcat from the shell i get:

```
$topcat
Can't find topcat classes in /usr/local/bin - looked for: ../lib/topcat/topcat.jar topcat-full.jar topcat-lite.jar ../Contents/Resources/Java/topcat-full.jar ../TOPCAT.app/Contents/Resources/Java/topcat-full.jar
```

The only way I managed to make topcat works is by:

```
$java -jar /usr/local/Cellar/topcat/4.0-1/share/java/topcat-full.jar
```

But it's not a very convenient solution.
Do have a better way to get topcat working like a more classical app after the installation with homebrew ?

**Comment by dpo** [comment]
This sounds like a Java question (with which I am not familiar). Isn't there some environment variable that points to paths to search for jar files?

**Comment by apjanke** [comment]
[Topcat](http://www.star.bris.ac.uk/~mbt/topcat/) doesn't have a traditional distribution. That downloadable `topcat-full.jar` file is a Java library. It needs a couple other bits to run as an application.

They provide a `topcat` command to go with that JAR file [as a separate download](http://www.star.bris.ac.uk/~mbt/topcat/topcat). If you grab that and put it together with the JAR, you'll be closer to running.

You could ask the author to put together a distribution if they want to provide Topcat as a standalone program: at a minimum, this would be something like a version-stamped tarball or zip file with the JAR file, that wrapper script, and installation instructions or an installer script. (The JAR file is currently being downloaded from some apparently unrelated person's SourceForge site because th downloads on http://www.star.bris.ac.uk/~mbt/topcat/ don't have versions in the file names.)

But if you look at [this FAQ Q1.13](http://www.star.bris.ac.uk/~mbt/topcat/faq.html#compileFromSource) about compiling topcat from source code, it points to a [GitHub repo for StarJava](https://github.com/Starlink/starjava/), the project Topcat was a part of, and the [topcat/ folder there](https://github.com/Starlink/starjava/tree/master/topcat/src) has a launcher script and more stuff you'd need. And they have tagged releases on that repo. For example, here's [topcat-4.3.1](https://github.com/Starlink/starjava/releases/tag/topcat-4.3-1).

The thing to do for Homebrew is probably to change the `topcat` formula to grab the source tarballs from that GitHub repo, build it from source, and do a complete installation from that. The JAR file on its own is inadequate, and those `caveats` aren't sufficient to get it running.

**Comment by apjanke** [comment]
How badly do you want to use this? It doesn't build out of the box on OS X for me when using the source from that GitHub repo. (Doesn't have a full build script, and doing the manual steps is failing.) How badly do you want to use this? I think we could get it building, but it would take a bit of work.

**Comment by iMichka** [comment]
There will be a version bump of topcat as soon as I merged this PR: #5713

The `java -jar` call is described on their website as the way to start the app (it's a .jar, so this is correct). There seem to be some activity in the project mentioned by @apjanke (https://github.com/Starlink/starjava/), so maybe something could be done on that side.

But this is obviously not something we are working on (we only have limited resources). Pull request are welcome though! Closing here.
